tslearn/tests/sklearn_patches.py

```diff
@@ -0,0 +1,507 @@
+from tslearn.generators import random_walks, random_walk_blobs
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance
+
+import sklearn
+from sklearn.base import clone
+from sklearn.utils.testing import *
+from sklearn.utils.estimator_checks import *
+from sklearn.utils.estimator_checks import (_yield_classifier_checks,
+                                            _yield_regressor_checks,
+                                            _yield_transformer_checks,
+                                            _yield_clustering_checks,
+                                            _yield_outliers_checks)
+try:
+    from sklearn.utils.estimator_checks import _yield_checks
+except ImportError:
+    from sklearn.utils.estimator_checks import _yield_non_meta_checks
+    _yield_checks = _yield_non_meta_checks
+
+from sklearn.metrics import adjusted_rand_score, accuracy_score
+from sklearn.model_selection import ShuffleSplit
+from sklearn.model_selection._validation import _safe_split
+from sklearn.pipeline import make_pipeline
+
+import warnings
+import numpy as np
+
+
+_DEFAULT_TAGS = {
+    'non_deterministic': False,
+    'requires_positive_X': False,
+    'requires_positive_y': False,
+    'X_types': ['2darray'],
+    'poor_score': False,
+    'no_validation': False,
+    'multioutput': False,
+    "allow_nan": False,
+    'stateless': False,
+    'multilabel': False,
+    '_skip_test': False,
+    'multioutput_only': False,
+    'binary_only': False,
+    'requires_fit': True}
+
+
+def _safe_tags(estimator, key=None):
+    # if estimator doesn't have _get_tags, use _DEFAULT_TAGS
+    # if estimator has tags but not key, use _DEFAULT_TAGS[key]
+    if hasattr(estimator, "_get_tags"):
+        if key is not None:
+            return estimator._get_tags().get(key, _DEFAULT_TAGS[key])
+        tags = estimator._get_tags()
+        return {key: tags.get(key, _DEFAULT_TAGS[key])
+                for key in _DEFAULT_TAGS.keys()}
+    if key is not None:
+        return _DEFAULT_TAGS[key]
+    return _DEFAULT_TAGS
+
+
+def _create_small_ts_dataset():
+    return random_walk_blobs(n_ts_per_blob=10, n_blobs=3, random_state=1,
+                             sz=50, noise_level=0.1)
+
+
+# Patch BOSTON dataset of sklearn to fix _csv.Error: line contains NULL byte
```

Comment:
Not only because of the bug: also because it is not a time series dataset so it makes little sense to test for it in tslearn :)